src/Helper/PropagateOptionsHelper.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Helper;
+
+use Yiisoft\Validator\RuleInterface;
+use Yiisoft\Validator\SkipOnEmptyInterface;
+use Yiisoft\Validator\SkipOnErrorInterface;
+use Yiisoft\Validator\WhenInterface;
+
+/**
+ * A helper class used to propagate options' values from a single parent rule to its single child rule.
+ */
+final class PropagateOptionsHelper
+{
+    /**
+     * Propagates options' values from a single parent rule to its single child rule. The following options' values are
+     * propagated:
+     *
+     * - `$skipOnEmpty` (both rules must implement {@see SkipOnEmptyInterface}).
+     * - `$skipOnError` (both rules must implement {@see SkipOnErrorInterface}).
+     * - `$when` (both rules must implement {@see WhenInterface}).
+     *
+     * @param RuleInterface $parentRule A parent rule which options' values need to be propagated.
+     * @param RuleInterface $childRule A child rule which options' values must be changed to be the same as in parent.
+     *
+     * @return RuleInterface A child rule with changed options' values or unchanged if none of the required interfaces
+     * were implemented.
+     */
+    public static function propagateForSingleRule(RuleInterface $parentRule, RuleInterface $childRule): RuleInterface
```

Comment:
```suggestion
    public static function propagate(RuleInterface $parentRule, RuleInterface ...$childRule): RuleInterface
```

Maybe use such a signature? It's allow remove a little more duplicate code.

Reply:
Could you show an example how it will be integrated in this case?

Reply:
https://github.com/yiisoft/validator/pull/471